### ISSUE TITLE
Add ability to define bundles of configurations that are not cross multiplied

### ISSUE THREAD

**Issue by panahiparham** [panahiparham/ml-experiment-definition#33]
It can be a useful addition to let the user pick if some configurations should not be cross multiplied. For an example use case imagine a previous version sweeps learning rates to find the best performing option. Now we are interested in the best learning rate for each batch size option. In current design, learning rate and batch sizes below are cross multiplied.
```
part = DefinitionPart('my-exp', base = './')

part.add_sweepable_property('learning_rate', [0.0001, 0.001, 0.01])
part.add_property('beta1', 0.9)
part.add_property('beta2', 0.999)
part.add_sweepable_property('batch_size', [4, 8])
```

I propose to add a new method to `DefinitionPart` called `add_property_zip` that adds only the corresponding values configurations
```
part = DefinitionPart('my-exp', base = './')

part.add_property('beta1', 0.9)
part.add_property('beta2', 0.999)
part.add_property_zip({
    'learning_rate': [0.01, 0.0001],
    'batch_size': [4, 8],
})
```

Now only two configurations are added where learning rate and batch size are `(0.01, 4)` and `(0.0001, 8)`.
